### ISSUE TITLE
Cell by cell parse feedback in Slim tables - similar to FitLibrary parse mechanism

### ISSUE THREAD

**Issue by brucetrask** [unclebob/fitnesse#654]
Hello,

FitLibrary provides a method for parsing individual cells and reporting the result of that parse in the cell of that param.  That provides cell by cell feedback on input params from the table to the fixture.  For example, if the fixture expects a float and a string is sent, the cell with the string in it will be yellow with the message "Invalid Number".  This is how FitLibrary 2.0 works.

I am not seeing anything similar in Slim.  For example,  http://www.fitnesse.org/FitNesse.FullReferenceGuide.UserGuide.WritingAcceptanceTests.SliM.DecisionTable
if one of the pints of milk values is changed from 0 to string like "asdf" then the entire row goes yellow with an expandable error section below the row which shows a stack trace.  I know we can modify that stack trace message with the "message:<<>>" mechanism but I want to have it reported on the offending cell only.

Is that possible in Slim?  Is it easy to extend Slim to support such a feature?

Thanks in advance.

Regards,
Bruce

**Comment by brucetrask** [comment]
Hello,

I have another example that illustrates what I am looking for.

On a FitLibrary DoFixture, one can have a Domain Object in the signature of the fixture function and the framework will call the parse function on the Domain Object so that the object can convert and or evaluate the validity of the param and throw an FitLibraryException if needed.  That will get rendered in the param cell of the input table with the error message.

On a Slim script table it does not work the same way.  If there is a Domain Object in the signature of the fixture and there is an added converter for that object that finds a problem with the input param and throws an exception in the case of a bad value. In the Slim Script table case, it does not get rendered in the cell of the input param but rather as an exception for the entire call.  

As an example of what I am talking about.  Let's take
http://www.fitnesse.org/FitNesse.UserGuide.WritingAcceptanceTests.SliM.ScriptTable
In the case of the second row 
| login with username | Bob | and password | xyzzy |
Let's we modify the signature of the fixture to be 
public boolean loginWithUsernameAndPassword(MyDomainObject userName, String password) {
...

and add a new Domain Object 
public class MyDomainObject {
}
and create the necessary converter
public class MyDomainObjectConverter implements fitnesse.slim.Converter<MyDomainObject>
{

	@Override
	public String toString(MyDomainObject o) {
		// TODO Auto-generated method stub
		System.out.println("call to toString of the MyDomainObjectConverter");
		
		return null;
	}

	@Override
	public MyDomainObject fromString(String arg) {
		// TODO Auto-generated method stub
		System.out.println("call to fromString of the DomainObjectConverter");
		if(true)  // simulate something wrong with the param
			throw new IllegalArgumentException("message:<<something wrong with the param>>");
		return null;
	}
}

the test will then show the second row as

|login with username something wrong with the param| 	Bob | and password |xyzzy|

In other words, unlike the DoFixture parse mechanism that gets rendered in the offending cell, the Script table converter mechanism results in a method failure that is not rendered in the offending cell.

**Comment by brucetrask** [comment]
I did not mean to close this one but rather just save my comment. sorry.  I would like to keep this one open

**Comment by amolenaar** [comment]
Solved by #682.
